### PR TITLE
Fix select all consent functionality

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
There's currently a race condition when using the select all or unselect all checkbox for consents.  This was fixed temporarily by throttling the individual updates for each of the (de)selected consents.  Unfortunately before this could be refactored, the throttle was accidentally removed.

This PR fixes that by sending all required updates in one ajax request.

## Screenshots
N/A

## What is the value of this and can you measure success?
We make the updates that users expect to be made to their consents!

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/identity/consents.js [hunk 1]
```diff
@@ -37,9 +37,6 @@ const UNSUBSCRIPTION_SUCCESS_MESSAGE =
     "You've been unsubscribed from all Guardian marketing newsletters and emails.";
 const ERR_MALFORMED_HTML = 'Something went wrong';
 
-const updateConsent = (consent: SettableConsent): Promise<void> =>
-    setConsent([consent]);
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/identity/consents.js]: I wasn't sure whether this added much over just inlining `setConsent([consent])` but happy to put back.

File: static/src/javascripts/projects/common/modules/identity/consents.js [hunk 2]
```diff
@@ -241,38 +238,51 @@ const bindCheckAllSwitch = (labelEl: HTMLElement): void => {
...
-                if (!(checkboxEl instanceof HTMLInputElement)) {
-                    throw new Error(ERR_MALFORMED_HTML);
-                }
                 checkboxEl.checked = getCheckedAllStatus(wrappedCheckboxEls);
                 titleEl.innerHTML = getTextForStatus(checkboxEl.checked);
                 labelEl.style.visibility = 'visible';
                 labelEl.style.pointerEvents = 'all';
             });
 
+        const getInputFieldsFromCheckboxes = checkboxesToUpdate =>
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/identity/consents.js]: Not sure about this method name.  It's actually retrieving an array of promises, which when complete will contain an array (node list) of input fields.

**Review by Mullefa — APPROVED** [review]
Looks good. Out of interest, is there an event listener that can be removed?

**Comment by jfsoul** [comment]
> is there an event listener that can be removed?
I think the event listener stays for when users click one of the consent checkboxes (rather than select all).

**Pull request merged**